CompanyEmployees.Core.Domain/Repositories/IAttendanceRepository.cs

```diff
@@ -0,0 +1,13 @@
+using CompanyEmployees.Core.Domain.Entities;
+
+namespace CompanyEmployees.Core.Domain.Repositories;
+
+public interface IAttendanceRepository
+{
+    Task<IEnumerable<Attendance>> GetAllAttendancesAsync(bool trackChanges, CancellationToken ct = default);
+    Task<Attendance> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default);
+    Task<IEnumerable<Attendance>> GetAttendancesByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default);
+    Task<Attendance?> GetTodayAttendanceByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default);
```

Comment:
Consider making the return type of GetAttendanceAsync nullable (Attendance?) to explicitly represent the potential for a null result if the attendance is not found. This aligns better with modern C# practices and improves clarity when handling non-existent records.

```suggestion
Task<Attendance?> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default);
```